GroceryList/app/src/main/java/com/keenanzucker/grocerylist/MainActivity.java

```diff
@@ -0,0 +1,150 @@
+
+                                // Remove the item within array at position
+                                items.remove(itemPos);
+                                // Then insert item at same location -- aka 'editing' the text
+                                itemsAdapter.insert(inputText, itemPos);
+                                // Refresh the adapter
+                                itemsAdapter.notifyDataSetChanged();
+
+                            }
+                        });
+
+                        //Negative button on dialog box -- "cancel"
+                        builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                dialog.cancel();
+                            }
+
+                        });
+
+                        //Show the dialog box!
+                        AlertDialog alertDialog = builder.create();
+                        alertDialog.show();
+                    }
+                }
+        );
+
+        // Long click list items to delete them
+        lvItems.setOnItemLongClickListener(
+                new AdapterView.OnItemLongClickListener() {
+                    @Override
+                    public boolean onItemLongClick(AdapterView<?> adapter,
+                                                   View item, int pos, long id) {
+                        // Remove the item within array at position
+                        items.remove(pos);
+                        // Refresh the adapter
+                        itemsAdapter.notifyDataSetChanged();
+                        // Return true consumes the long click event (marks it handled)
+                        return true;
+                    }
+                }
+        );
+    }
+
+    // Add the items when the button is clicked!
+    public void onAddItem(View v) {
+        EditText etNewItem = (EditText) findViewById(R.id.etNewItem);
+        String itemText = etNewItem.getText().toString();
+        itemsAdapter.add(itemText);
+        etNewItem.setText("");
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+}
```

Comment:
Great code. Bottom line, cut out some of the unnecessary white space and comments and just add an overarching comment at the top of each method.